[souledfigurine] iP

souledfigurine opened the pull request:
# Baymax
> "it is alright to cry." -Baymax

**Baymax** is your friendly **task management chatbot**, designed to help you keep track of your tasks effortlessly. With Baymax, you can:
- [x] Create tasks (To-do, Deadlines, Events) 📝 
- [x] Mark tasks as done ❌  
- [x] Unmark tasks
- [x] Delete tasks
- [x] List tasks
- [x] Find tasks quickly
- [ ] Save all tasks automatically

It is:
- text-based
- easy to learn
- Super fast to use!

All you need to do is,
1. Download it from 
2. double-click it
3. add your tasks
4. let it manage your tasks for you!

And it is ~~$0.99~~ *FREE*!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

souledfigurine committed:
Level-0: Rename, Greet, Exit

souledfigurine committed:
added scanner to level 0

souledfigurine committed:
echo commands + exit when user types 'bye'

souledfigurine committed:
included addTask, String[],

souledfigurine committed:
added printList()

souledfigurine committed:
Made the list numbered

souledfigurine committed:
created boolean[] completed, and markAsDone()

souledfigurine committed:
updated addTask to ensure encapsulation

souledfigurine committed:
markAsDone successfully implemented, updated printList

souledfigurine committed:
unmark() successfully created and implemented

souledfigurine committed:
new Task.class

souledfigurine committed:
implemented Task.java into baymax.java

souledfigurine committed:
created todo.java, event.java, deadline.java

souledfigurine committed:
implemented deadline, event and todo into baymax

souledfigurine committed:
created exception class, not sure how to do exception

souledfigurine committed:
failed attempt at text-ui-testing

souledfigurine committed:
inserted delete function

souledfigurine committed:
added commments to justify error handling

souledfigurine committed:
updated expected.txt to ensure no difference

souledfigurine committed:
complete level 7

souledfigurine committed:
complete level 7

souledfigurine committed:
merge level-7 into master

zephaniahong commented on src/main/java/Baymax.java:
```diff
@@ -0,0 +1,163 @@
+                break;
+            }
+            //Check for list
+            if (userInput.equalsIgnoreCase("list")) {
```
Perhaps you could consider using else if statements? This would remove the need for break and continue statements.

zephaniahong commented on src/main/java/Baymax.java:
```diff
@@ -0,0 +1,163 @@
+
+    private void unmark(int taskNumber) {
+        //Error2: Handled invalid unmark function
+        if (!taskList.get(taskNumber - 1).getIsCompleted()) {
```
I think `isCompleted` should suffice
```suggestion
        if (!taskList.get(taskNumber - 1).isCompleted()) {
```

DengTianle replied:
Agree. Perhaps `isCompleted` is more intuitive for the method name?

zephaniahong commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,70 @@
+import task.*;
+
+public class Storage {
+    private static final String FILE_PATH = "./data/duke.txt";
```
Great use of a static variable!

zephaniahong commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,70 @@
+            Boolean isCompleted = parts[1].equals("true");
+
+            switch (type) {
+                case "T":
+                    return new Todo(parts[2], isCompleted);
+                case "D":
+                    return new Deadline(parts[2], parts[3], isCompleted);
+                case "E":
+                    return new Event(parts[2], parts[3], parts[4], isCompleted);
```
There shouldn't be indentation for case clauses

zephaniahong commented on src/main/java/Baymax.java:
```diff
@@ -0,0 +1,163 @@
+                continue;
+            }
+            //add Todo
+            if (userInput.startsWith("todo ")) {
+                baymax.addTodo(userInput.substring(5));
+                continue;
+            }
+            //add Deadline
+            if (userInput.startsWith("deadline ")) {
+                String[] parts = userInput.substring(9).split(" /by ");
+                baymax.addDeadline(parts[0], parts[1]);
+                continue;
+            }
+            //add Event
+            if (userInput.startsWith("event ")) {
```
Is there a reason why theses 3 commands have whitespace after them? You can consider stripping the string prior.

zephaniahong reviewed (commented):
Overall, I think your code looks good! Just a couple of nits. Keep it up!

DengTianle commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+package task;
+
+public abstract class Task {
+    protected String Name;
```
Perhaps it should be `name` as per the convention?

DengTianle reviewed (commented):
PR review 2 (back-up)
I like that your code is generally easy to understand, good job!

souledfigurine committed:
Add LocalDateTime to Deadline.java

souledfigurine committed:
change String LocalSate format

souledfigurine committed:
Merge branch 'branch-Level-8'

souledfigurine committed:
creates Tasklist.java

souledfigurine committed:
Creates Parser.java

souledfigurine committed:
creats Ui.java

souledfigurine committed:
Updates Baymax.java to match oop

souledfigurine committed:
updates Storage.jave and creates AddCommand.java, AddTodoCommand.java, AddEventCommand.java, AddDeadlineCommand.java

souledfigurine committed:
creats DeleteCommand.java

souledfigurine committed:
creates command classes

souledfigurine committed:
puts command and task package into baymax package and creats mark and unmarkCommand

souledfigurine committed:
Merge branch 'branch-A-MoreOOP'

souledfigurine committed:
Merge branch 'add-gradle-support'

souledfigurine committed:
adds UiTest.java and DeadlineTest.java

souledfigurine committed:
updates build.gradle

souledfigurine committed:
Includes JavaDoc

souledfigurine committed:
changes case indentation and 1 variable to follow coding standard

souledfigurine committed:
Creates FindCommand.java and updates Parser.java, TaskList.java and Ui.java to find keywords

souledfigurine committed:
Merge branch 'branch-A-JavaDoc'

souledfigurine committed:
Resolves merge conflict

souledfigurine committed:
merges conflict

souledfigurine committed:
adds JavaDoc

souledfigurine committed:
Adds checkstyle

souledfigurine committed:
changes print comment form printExit in Ui.java

souledfigurine committed:
Merge tag 'branch-A-CheckStyle'

souledfigurine committed:
updates build.gradle to include javaFx dependencies as well as created resources directory

souledfigurine committed:
imports relevant classes from javaFx tutorial. This includes DialogBox.java, Duke.java, Launcher.java, Main.java and MainWindow.java

souledfigurine committed:
Updates Main.java, MainWindow.java and Baymax.java so that they now all interact with each other rather than with duke.java

souledfigurine committed:
implements GUI successfully. Removes UiTest.java, Ui.java, ExitCommand.java, Duke.java

souledfigurine committed:
Merge branch 'branch-Level-10'

souledfigurine committed:
renames Images folder to images

souledfigurine committed:
Add asserts to Parser.java, TaskList.java, Deadline.java and Event.java

souledfigurine committed:
Add two JUnit tests, ParserTest.java and EventTest.java

souledfigurine committed:
Add named constants to execute function in command

souledfigurine committed:
Merge branch 'master' into branch-A-Assertions

souledfigurine committed:
Merge pull request #2 from souledfigurine/branch-A-Assertions

Add asserts to Parser.java, TaskList.java, Deadline.java and Event.java

souledfigurine committed:
Merge branch 'master' into branch-A-CodeQuality

souledfigurine committed:
Add HelpCommand.java to create list of commands available for user

souledfigurine committed:
Merge branch 'branch-C-HelpCommand'

souledfigurine committed:
Make commands case-insensitive in Parser.java

souledfigurine committed:
Fix exception handling for incorrect datetime format. Make command case-insensitive

souledfigurine committed:
Make commands /by /from and /to case-insensitive

souledfigurine committed:
Update README.md to include description of functional features

souledfigurine committed:
Insert Ui.png into docs

souledfigurine committed:
Remove imports from AddDeadlineCommand.java

There are several unused imports in AddDeadlineCommand.java

Let's clean up the clutter in the code.

souledfigurine committed:
Remove import from DeadlineTest.java

Let's tidy remove unused imports to make our code neater.

souledfigurine committed:
Fix Tasklist.java

souledfigurine committed:
Delete Duke README.md
